stock_by_warehouse/models/product_product.py

```diff
@@ -108,3 +115,69 @@ def _compute_get_quantity_warehouses_json(self):
                 product.qty_available - product.outgoing_qty
             })
         return json.dumps(info)
+
+    @api.multi
+    def _compute_get_stock_location(self):
+        """Method to get the stock quants that have a specific product and the locations of all the warehouses.
+        It will return a dict with the location and quantity, the location with the most quantity of the product,
+        and a number of locations where the product is available.
+        """
+
+        # Get original from onchange
+        self_origin = self._origin if hasattr(self, '_origin') else self
+        info = {'title': _('Stock by Warehouse and Locations'), 'content': []}
+        if not self_origin.exists():
+            return json.dumps(info)
+
+        self_origin.ensure_one()
```

Comment:
Why this?

Reply:
It is used to avoid that multiple records call this method when it is called from another method other than an onchange and now I add this explanation on the code.